src/apify_client/clients/resource_clients/key_value_store.py

```diff
@@ -0,0 +1,152 @@
+
+        Args:
+            limit (int, optional): Number of keys to be returned. Maximum value is 1000
+            exclusive_start_key (str, optional): All keys up to this one (including) are skipped from the result
+        """
+        request_params = self._params(
+            limit=limit,
+            exclusiveStartKey=exclusive_start_key,
+        )
+
+        response = self.http_client.call(
+            url=self._url('keys'),
+            method='GET',
+            params=request_params,
+        )
+
+        return _parse_date_fields(_pluck_data(response.json()))
+
+    def get_record(self, key: str, *, as_bytes: bool = False, as_file: bool = False) -> Optional[Dict]:
+        """Retrieves the given record from the key-value store.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/record/get-record
+
+        Args:
+            key (str): Key of the record to retrieve
+            as_bytes (bool, optional): Whether to retrieve the record as unparsed bytes, default False
+            as_file (bool, optional): Whether to retrieve the record as a file-like object, default False
+        """
+        try:
+            if as_bytes and as_file:
+                raise ValueError('You cannot have both as_bytes and as_file set.')
+
+            response = self.http_client.call(
+                url=self._url(f'records/{key}'),
+                method='GET',
+                params=self._params(),
+                stream=as_file,
+                parse_response=(not as_bytes and not as_file),
+            )
+
+            return {
+                'key': key,
+                'value': response._maybe_parsed_body,  # type: ignore
+                'content_type': response.headers['content-type'],
+            }
+
+        except ApifyApiError as exc:
+            _catch_not_found_or_throw(exc)
+
+        return None
+
+    def set_record(self, key: str, value: Any, content_type: str = None) -> None:
+        """Sets a value to the given record in the key-value store.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/record/put-record
+
+        Args:
+            key (str): The key of the record to save the value to
+            value (Any): The value to save into the record
+            content_type (str, optional): The content type of the saved value
+        """
+        # TODO revisit this when it's decided whether we keep using the signed URL route or not
+
+        headers = None
+
+        if not content_type:
+            if _is_file_or_bytes(value):
+                content_type = 'application/octet-stream'
+            elif isinstance(value, str):
+                content_type = 'text/plain; charset=utf-8'
+            else:
+                content_type = 'application/json; charset=utf-8'
+
+        if 'application/json' in content_type and not _is_file_or_bytes(value) and not isinstance(value, str):
+            value = json.dumps(value, indent=2)
```

Comment:
using `indent=...` actually makes JSON larger. Since we are sending this over network, let's do it without it unless there is good reason to keep it.

Reply:
I added it so that the behavior between the JS and Python client is the same, it improves readability when displayed in the app and if we `gzip` the POST requests (which we're currently not doing in this client, but it's on the roadmap), the size difference should not be too big. But I'm not hell-bent on keeping it, what do you think, @mnmkng

Reply:
Please, at least add big TODO here. From my previous experience the difference was couple of X for large JSONs. Thx